internal/provider/resource_integration_test.go

```diff
@@ -265,6 +265,30 @@ func TestAcc_ResourceIntegrationWithMultipleConsumers(t *testing.T) {
 	})
 }
 
+func TestAcc_ResourceIntegrationWithMultipleIntegrationsSameEndpoint(t *testing.T) {
+	if testingCloud != LXDCloudTesting {
+		t.Skip(t.Name() + " only runs with LXD")
+	}
+	srcModelName := acctest.RandomWithPrefix("tf-test-integration-offering")
+	dstModelName := acctest.RandomWithPrefix("tf-test-integration-consuming")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceIntegrationMultipleIntegrationsSameEndpoint(srcModelName, dstModelName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_integration.this", "model", dstModelName),
+					resource.TestCheckResourceAttr("juju_integration.this", "id", fmt.Sprintf("%v:%v:%v", dstModelName, "apptwo:source", "appzero:sink")),
+					resource.TestCheckResourceAttr("juju_integration.this2", "model", dstModelName),
+					resource.TestCheckResourceAttr("juju_integration.this2", "id", fmt.Sprintf("%v:%v:%v", dstModelName, "apptwo:source", "appone:sink")),
+				),
```

Comment:
This test doesn't seem to be proving the fact that we will only delete 1 of the 2 integrations.

Reply:
Ah the issue is not that we only want to delete 1 of the 2 but rather that we can't delete because of the Juju error, "ambiguous relation".